### PR TITLE
Allow external configuration

### PR DESCRIPTION
**Pull request by rodrigohenriques**
Create `AutocompleteOption` to expose a way to customize queries.

### CODE REVIEW

File: library/src/main/kotlin/com/a99/rxplaces/options/AutocompleteOptions.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    internal var offset: Int? = null
+    internal var location: Location? = null
+    internal var radius: Int? = null
+    internal var language: String? = null
+    internal var types: List<AutocompleteType> = listOf()
+    internal var components: List<String> = listOf()
+    internal var strictBounds: Boolean? = null
+
+    fun offset(data: Builder.() -> Int) = apply { offset = data() }
```
Comment (rsicarelli) [inline library/src/main/kotlin/com/a99/rxplaces/options/AutocompleteOptions.kt]: Can you please improve this `data` function name? Something like `getOffset()` or simply `offset()`. 

Please apply this for all functions below

File: library/src/test/java/com/a99/rxplaces/ExtensionsKtTest.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import com.a99.rxplaces.options.AutocompleteType
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.CoreMatchers.not
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Test
+import java.util.Random
+
+class ExtensionsKtTest {
+  @Test
+  fun formatComponents_emptyList() {
```
Comment (rsicarelli) [inline library/src/test/java/com/a99/rxplaces/ExtensionsKtTest.kt]: We should improve the test names. When reading those tests, it is not clear what the desired behaviour for the extensions

File: library/src/main/kotlin/com/a99/rxplaces/Extensions.kt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.a99.rxplaces
+
+import android.location.Location
+
+internal fun <T> List<T>.format(): String? {
```
Comment (rsicarelli) [inline library/src/main/kotlin/com/a99/rxplaces/Extensions.kt]: Please provide a more described name for this one, such as `insertPipeBetweenValues()` or something like that

File: library/src/test/java/com/a99/rxplaces/AutocompleteTypeTest.kt [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  fun test_toString() {
+    // when
+    val cities = AutocompleteType.CITIES.toString()
+    val regions = AutocompleteType.REGIONS.toString()
+    val establishment = AutocompleteType.ESTABLISHMENT.toString()
+    val address = AutocompleteType.ADDRESS.toString()
+    val geocode = AutocompleteType.GEOCODE.toString()
+
+    // then
+    assertThat(cities, `is`("(cities)"))
```
Comment (rsicarelli) [inline library/src/test/java/com/a99/rxplaces/AutocompleteTypeTest.kt]: Can you please create another test for the case when we are inserting `()` in the parameters?

File: library/src/main/kotlin/com/a99/rxplaces/Extensions.kt (hunk 3 above)
Comment (walmyrcarvalho) [inline library/src/main/kotlin/com/a99/rxplaces/Extensions.kt]: Indeed, `format` is a too generic name.

File: library/src/main/kotlin/com/a99/rxplaces/options/AutocompleteOptions.kt [hunk 5]
```diff
@@ -0,0 +1,54 @@
+package com.a99.rxplaces.options
+
+import android.location.Location
+
+class AutocompleteOptions private constructor(
```
Comment (walmyrcarvalho) [inline library/src/main/kotlin/com/a99/rxplaces/options/AutocompleteOptions.kt]: TIL that Autocomplete is a valid english word. :O

File: library/src/test/java/com/a99/rxplaces/ExtensionsKtTest.kt [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    // given
+    val limit = Random().nextInt(10000)
+    val range = 1..limit
+    val components = range.map { "country:$it" }
+
+    // when
+    val formatted = components.format()
+
+    // then
+    assertThat(formatted!!.last().toString(), not(`is`("|")))
```
Comment (walmyrcarvalho) [inline library/src/test/java/com/a99/rxplaces/ExtensionsKtTest.kt]: I would try to not use `!!` for tests, we can't assume that we'll have 100% not null scenarios for this case.

Comment (rodrigohenriques) [inline library/src/test/java/com/a99/rxplaces/ExtensionsKtTest.kt]: My function is nullable so in that specific scenario I want to throw an exception if my function returns null.

File: library/src/main/kotlin/com/a99/rxplaces/Extensions.kt [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+import android.location.Location
+
+internal fun <T> List<T>.toPipedString(): String? {
+  if (isEmpty()) return null
+
+  return map { it.toString() }
+      .reduce { accumulator, next -> accumulator.plus("|").plus(next) }
+}
+
+internal fun Location.formatWithComma() : String {
```
Comment (walmyrcarvalho) [inline library/src/main/kotlin/com/a99/rxplaces/Extensions.kt]: Suggestion:

`internal fun Location.formatWithComma() : String = "$latitude,$longitude"`

File: library/src/main/kotlin/com/a99/rxplaces/Extensions.kt [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package com.a99.rxplaces
+
+import android.location.Location
+
+internal fun <T> List<T>.toPipedString(): String? {
```
Comment (stik3nd) [inline library/src/main/kotlin/com/a99/rxplaces/Extensions.kt]: Hey, checkout `joinToString` method from Kotlin standard library.

Comment (rodrigohenriques) [inline library/src/main/kotlin/com/a99/rxplaces/Extensions.kt]: Yeah, sure. Feel free to create PRs with this sort of changes.

**Pull request merged**